src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -490,16 +491,19 @@ private bool RemoveFromPage(bool allowRecurse, out long oldValue)
                 _internalCursor._pos > 0 && // nothing to do for a single leaf node
                 state.Header->FreeSpace > Constants.Storage.PageSize / 3)
             {
+                // If there is less than half of the size available, it is in our best interest to defrag the receiving page to avoid having to
+                // try again and again without achieving any gain.
+                if (state.Header->Upper - state.Header->Lower < state.Header->FreeSpace / 2)
+                    DefragPage(_llt, ref state);
+
                 MaybeMergeEntries(ref state);
             }
 
             return true;
         }
-
-        private void Verify()
+        
+        private void Verify(ref CursorState current)
```

Comment:
That should be `[Conditional]` ?

Reply:
No, most likely if you need to use it to verify you dont want to run in DEBUG

Reply:
Who is calling this? That can be expensive

Reply:
Nobody. It is a debugging resource.